src/lib/notifications.ts

```diff
@@ -0,0 +1,92 @@
+import { AppError } from 'src/lib/errors'
+import { randomBytes } from 'crypto'
+
+interface NotificationMessage {
+  title: string
+  message?: string
+}
+
+/**
+ * A notification is either dismissible or not
+ * if it is dismissible it can also optionally have
+ * - a label (for the dismiss button)
+ * - a timeout (to auto-dismiss)
+ * - a callback
+ */
+
+type NotificationDismiss = {
+  dismiss?: boolean | {
```

Comment:
Can this be typed as:

```
type NotificationDismiss = {
  dismiss?: false | {
    label?: string,
    timeout?: number,
    onDismiss?: (...args: any) => void
  },
}
```
This would allow you to avoid the `dismiss && typeof dismiss === 'object'`.

Or is there a case when dismiss will be set to `true` (I couldn't any in the code)?

Reply:
:+1: done